### DISCUSSION TITLE
[QUESTION] question about iptables and ip route on android

### DISCUSSION THREAD

**Discussion by Beat-YT** [Mygod/VPNHotspot#461] [Q&A]
Hello, I am trying to get Wifi-Calling to go through my VPN.

I tried to use iptables to do so, without success.

Wifi calling uses uid 1001 and udp port 500 and 4500.

I tried this:

Create a table that is used when marked by 2

ip rule add fwmark 2 table 3

Set what the table do. Make everything that falls under the rule go through the VPN (tun0)

ip route add 0.0.0.0/0 via 10.8.0.26 dev tun0 table 3

Then the iptable rule to mark everything from udp port 500 to tun0:

iptables -t nat -A PREROUTING -i wlan0 -p udp --dport 500 -j DNAT --to-destination 10.8.0.26:500
iptables -A FORWARD -i wlan0 -o tun0 -p udp --dport 500 -j ACCEPT
iptables -t nat -A POSTROUTING -o tun0 -p udp --dport 500 -j SNAT --to-source 10.8.0.26

this is obviously not working.

**Comment by Beat-YT** [comment]
![Screenshot_20230124-203206.jpg](https://user-images.githubusercontent.com/66485277/214460357-988b36f3-9d2a-4dfc-bc53-2a9d37f02545.jpg)

**Reply by Mygod** [reply]
It looks like you should bypass the Phone app in your VPN app.

**Reply by Beat-YT** [reply]
But I’m trying to get to go thorough the VPN

**Reply by Mygod** [reply]
Your VPN should support UDP first.

**Reply by Beat-YT** [reply]
> Your VPN should support UDP first.

You don’t get it, normally wifi calling does not pass through the VPN and I want to make it pass through. I need an ip table rule or something like that
